diff --git third-party/webscalesqlclient/src/mysys/stacktrace.c third-party/webscalesqlclient/src/mysys/stacktrace.c
index e915698..2fc8bb9 100644
--- third-party/webscalesqlclient/src/mysys/stacktrace.c
+++ third-party/webscalesqlclient/src/mysys/stacktrace.c
@@ -355,7 +355,7 @@ void my_print_stacktrace(uchar* stack_bottom, ulong thread_stack)
 #endif  /* __alpha__ */
 
   /* We are 1 frame above signal frame with NPTL and 2 frames above with LT */
-  sigreturn_frame_count = thd_lib_detected == THD_LIB_LT ? 2 : 1;
+  sigreturn_frame_count = 2;
 
   while (fp < (uchar**) stack_bottom)
   {
