diff --git source/Host/linux/Host.cpp source/Host/linux/Host.cpp
index cd2d481..4329e8a 100644
--- source/Host/linux/Host.cpp
+++ source/Host/linux/Host.cpp
@@ -14,7 +14,7 @@
 #include <sys/stat.h>
 #include <dirent.h>
 #include <fcntl.h>
-#ifndef __ANDROID__
+#if !defined(__ANDROID__) && !defined(__MUSL__)
 #include <execinfo.h>
 #endif
 
@@ -380,7 +380,7 @@ Host::GetProcessInfo (lldb::pid_t pid, ProcessInstanceInfo &process_info)
 void
 Host::Backtrace (Stream &strm, uint32_t max_frames)
 {
-#ifndef __ANDROID__
+#if !defined(__ANDROID__) && !defined(__MUSL__)
     if (max_frames > 0)
     {
         std::vector<void *> frame_buffer (max_frames, NULL);
@@ -394,8 +394,10 @@ Host::Backtrace (Stream &strm, uint32_t max_frames)
             ::free (strs);
         }
     }
-#else
+#elif defined __ANDROID__
     assert(false && "::backtrace() not supported on Android");
+#elif defined __MUSL__
+    assert(false && "::backtrace() not supported on musl libc");
 #endif
 }
 
