diff --git source/Plugins/Process/Linux/NativeProcessLinux.cpp source/Plugins/Process/Linux/NativeProcessLinux.cpp
index 56330e0..75a209d 100644
--- source/Plugins/Process/Linux/NativeProcessLinux.cpp
+++ source/Plugins/Process/Linux/NativeProcessLinux.cpp
@@ -68,7 +68,7 @@
 #include "ProcFileReader.h"
 #include "Plugins/Process/POSIX/ProcessPOSIXLog.h"
 
-#ifdef __ANDROID__
+#if defined(__ANDROID__) || defined(__MUSL__)
 #define __ptrace_request int
 #define PT_DETACH PTRACE_DETACH
 #endif
diff --git source/Plugins/Process/Linux/ProcessMonitor.cpp source/Plugins/Process/Linux/ProcessMonitor.cpp
index 3efeb5d..0414be0 100644
--- source/Plugins/Process/Linux/ProcessMonitor.cpp
+++ source/Plugins/Process/Linux/ProcessMonitor.cpp
@@ -51,7 +51,7 @@
 #include "Plugins/Process/POSIX/ProcessPOSIXLog.h"
 #include "ProcessMonitor.h"
 
-#ifdef __ANDROID__
+#if defined(__ANDROID__) || defined(__MUSL__)
 #define __ptrace_request int
 #define PT_DETACH PTRACE_DETACH
 #endif
