diff --git CMakeLists.txt CMakeLists.txt
index c5d8592..31b5840 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -460,10 +460,6 @@ ADD_SUBDIRECTORY(include)
 ADD_SUBDIRECTORY(spectrum)
 ADD_SUBDIRECTORY(backends)
 ADD_SUBDIRECTORY(tests)
-if (NOT WIN32)
-	ADD_SUBDIRECTORY(spectrum_manager)
-#	ADD_SUBDIRECTORY(spectrum2_send_message)
-endif()
 
 if (CPPUNIT_FOUND)
 	message("tests             : yes")
